test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -3372,6 +3372,48 @@ def forward(self, x, k):
         k = torch.tensor(3)
         self.run_test(MyModuleDynamic(), [x, k])
 
+    @disableScriptTest()
+    def test_auto_grad(self):
+        class MyClip(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, input, min):
+                ctx.save_for_backward(input)
+                return input.clamp(min=min)
+
+        class MyRelu(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, input):
+                ctx.save_for_backward(input)
+                return input.clamp(min=0)
+
+        from torch.onnx.symbolic_helper import _unimplemented
+
+        def symbolic_pythonop(g: torch._C.Graph, n: torch._C.Node, *args, **kwargs):
+            name = kwargs["name"]
+            if name == "MyClip":
+                return g.op("Clip", args[0], args[1])
+            elif name == "MyRelu":
+                return g.op("Relu", args[0])
+            else:
+                return _unimplemented("prim::PythonOp", "unknown node kind: " + name)
+
+        register_custom_op_symbolic("prim::PythonOp", symbolic_pythonop, 1)
+
+        class MyClipModule(torch.nn.Module):
+            def forward(self, x, min):
+                return MyClip.apply(x, min)
+        x = torch.randn(3, 3)
+        min = torch.tensor([0.0])
+        self.run_test(MyClipModule(), (x, min))
+
+        class MyReluModule(torch.nn.Module):
+            def forward(self, x):
+                return MyRelu.apply(x)
+        x = torch.randn(3, 3)
+        self.run_test(MyReluModule(), x)
+
+        unregister_custom_op_symbolic("prim::PythonOp", 1)
```

Comment:
Rather than call unregister at the end of the test, can we add code to unregister all custom ops in `tearDown`?
That way each test is guaranteed to start fresh.

Reply:
People add `register_custom_op_symbolic` for their own unit tests, but this registration is a global one. For example, one `aten::embedding` unit test actually replace the standard symoblic with the custom one. If we don't unregister, it will make other unit test not correct. So I would suggest make the unit test self-contained, so unregister each time.

Reply:
If we unregister **all** custom ops in `tearDown()` as I suggested, that will make each test self contained without each test having to worry about unregistering.

Reply:
In the current logic, we do not distinguish whether the op is custom or not after registering. Therefore we could not just unregister all "custom" ops. We need specify the op name that we want to unregister.

Reply:
Good point. We should fix that.
For now, please change to move the unregister immediately after the register, so it happens even if the test fails:

```python
register_custom_op_symbolic("prim::PythonOp", symbolic_python_op, 1)
self.addCleanup(unregister_custom_op_symbolic, "prim::PythonOp", 1)
```

Reply:
A little confused about this: If we unregister immediatly after register, then register custom op does not make effect, then the unit test would not work...